.github/workflows/test.yml

```diff
@@ -17,24 +17,38 @@ defaults:
     shell: bash -l {0}
 
 jobs:
-  detect-ci-trigger:
-    name: Check for upstream trigger phrase
+  detect-upstream-trigger:
+    name: Check for upstream trigger phrase to test with upstream dask
     runs-on: ubuntu-latest
     if: github.repository == 'dask-contrib/dask-sql'
     outputs:
-      triggered: ${{ steps.detect-trigger.outputs.trigger-found }}
+      triggered: ${{ steps.detect-upstream-trigger.outputs.trigger-found }}
     steps:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 2
       - uses: xarray-contrib/ci-trigger@v1.2
-        id: detect-trigger
+        id: detect-upstream-trigger
         with:
           keyword: "[test-upstream]"
+  detect-datafusion-upstream-trigger:
+    name: Check for upstream trigger phrase to test with upstream datafusion
+    runs-on: ubuntu-latest
+    if: github.repository == 'dask-contrib/dask-sql'
+    outputs:
+      triggered: ${{ steps.detect-datafusion-upstream-trigger.outputs.trigger-found }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - uses: xarray-contrib/ci-trigger@v1.2
+        id: detect-datafusion-upstream-trigger
+        with:
+          keyword: "[test-df-upstream]"
```

Comment:
Since these are fairly quick steps, wonder if it would make sense to consolidate this into a single job with two steps?